great_expectations/experimental/datasources/data_asset/data_connector/util.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+import copy
+import logging
+import re
+from typing import TYPE_CHECKING, Dict, List, Optional, Set
+
+from great_expectations.core.id_dict import IDDict
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+logger = logging.getLogger(__name__)
+
+try:
+    from azure.storage.blob import BlobPrefix
+except ImportError:
+    BlobPrefix = None
+    logger.debug(
+        "Unable to load azure types; install optional Azure dependency for support."
+    )
+
+try:
+    from google.cloud import storage
+except ImportError:
+    storage = None
+    logger.debug(
+        "Unable to load GCS connection object; install optional Google dependency for support"
+    )
+
+try:
+    import pyspark
+    import pyspark.sql as pyspark_sql
+except ImportError:
+    pyspark = None
+    pyspark_sql = None
+    logger.debug(
+        "Unable to load pyspark and pyspark.sql; install optional Spark dependency for support."
+    )
+
+
+def batch_definition_matches_batch_request(
+    batch_definition: BatchDefinition,
+    batch_request: BatchRequest,
+) -> bool:
+    if not (
+        batch_request.datasource_name == batch_definition.datasource_name
+        and batch_request.data_asset_name == batch_definition.data_asset_name
+    ):
+        return False
+
+    if batch_request.options:
+        for key in batch_request.options.keys():
+            if not (
+                key in batch_definition.batch_identifiers
+                and batch_definition.batch_identifiers[key]
+                == batch_request.options[key]
+            ):
+                return False
+
+    return True
+
+
+def map_data_reference_string_to_batch_definition_list_using_regex(
+    datasource_name: str,
+    data_connector_name: str,
+    data_reference: str,
+    regex_pattern: re.Pattern,
+    data_asset_name: str,
+) -> List[BatchDefinition] | None:
+    batch_identifiers: Optional[
+        IDDict
+    ] = convert_data_reference_string_to_batch_identifiers_using_regex(
+        data_reference=data_reference,
+        regex_pattern=regex_pattern,
+    )
+    if batch_identifiers is None:
+        return None
+
+    # Importing at module level causes circular dependencies.
+    from great_expectations.core.batch import BatchDefinition
+
+    return [
+        BatchDefinition(
+            datasource_name=datasource_name,
+            data_connector_name=data_connector_name,
+            data_asset_name=data_asset_name,
+            batch_identifiers=IDDict(batch_identifiers),
+        )
+    ]
+
+
+def convert_data_reference_string_to_batch_identifiers_using_regex(
```

Comment:
This should be called `batch_metadata`, there are no batch_identifiers in ZEP

Reply:
@donaldheppner Yes; but we still use `BatchDefinition` in ZEP and set `batch_metadata` to its `batch_identifiers` field.  Change this convenience method right now would be semantically wrong.  How about we make it a private method for now -- note that, fittingly, this method in ZEP is different from its V3 counterpart.  Happy to discuss.  Thanks.

Reply:
Agree, this should be made private (it is only required by the method above); it should also be moved into the FilePathDataConnector along with the above